catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/IngestCommand.java

```diff
@@ -538,4 +622,77 @@ private void submitToCatalog(ScheduledExecutorService batchScheduler,
             }
         }, 100, 100, TimeUnit.MILLISECONDS);
     }
+
+    private int getFilesCount(File rootFile) throws IOException {
+        if (!rootFile.isDirectory()) {
+            return 1;
+        }
+        final AtomicInteger count = new AtomicInteger(0);
+
+        Files.walkFileTree(rootFile.toPath(), new SimpleFileVisitor<Path>() {
+            @Override
+            public FileVisitResult visitFile(Path filePath, BasicFileAttributes attrs)
+                    throws IOException {
+
+                File file = filePath.toFile();
+                if (!file.isDirectory()) {
+                    count.incrementAndGet();
+                }
+                return FileVisitResult.CONTINUE;
+            }
+        });
+        return count.get();
+    }
+
+    private Map<String, List<File>> generateFileMap() throws IOException {
+        File inputFile = new File(filePath);
+        if (!inputFile.isDirectory()) {
+            inputFile = new File(inputFile.getParent() + CONTENT_PATH);
```

Comment:
Could you use https://docs.oracle.com/javase/7/docs/api/java/io/File.html#File%28java.io.File,%20java.lang.String%29 instead?